packages/react-storage/src/components/StorageImage/__tests__/StorageImage.test.tsx

```diff
@@ -56,42 +60,93 @@ describe('StorageImage', () => {
     expect(img).toHaveClass(className);
   });
 
-  it('should throw an error if both `imgKey` and `path` props are passed in', () => {
-    errorSpy.mockImplementation(() => {});
-
+  it('should throw an error if neither `imgKey` or `path` props are provided', () => {
     expect(() =>
       render(
-        // @ts-expect-error testing invalid input
-        <StorageImage
-          alt="StorageImage"
-          path={path}
-          imgKey={imgKey}
-          onGetUrlError={onError}
-        />
+        // @ts-expect-error force invalid input
+        <StorageImage alt="StorageImage" onGetUrlError={onError} />
       )
-    ).toThrow('StorageImage cannot have both imgKey and path props.');
+    ).toThrow(MISSING_REQUIRED_PROP_MESSAGE);
   });
 
-  it('should show console warning if `imgKey` prop is passed in', () => {
-    warnSpy.mockImplementation(() => {});
+  it('should show a console warning if both `imgKey` and `path` props are provided', () => {
     const { rerender } = render(
       <StorageImage alt="StorageImage" path={path} onGetUrlError={onError} />
     );
 
     expect(warnSpy).toHaveBeenCalledTimes(0);
 
     rerender(
+      // @ts-expect-error force invalid input
+      <StorageImage
+        alt="StorageImage"
+        imgKey={imgKey}
+        path={path}
+        onGetUrlError={onError}
+      />
+    );
+
+    expect(warnSpy).toHaveBeenCalledTimes(1);
+    expect(warnSpy).toHaveBeenCalledWith(HAS_PATH_AND_KEY_MESSAGE);
+  });
+
+  it('should show a console warning if a `imgKey` prop is provided', () => {
+    render(
+      <StorageImage
+        alt="StorageImage"
+        accessLevel={accessLevel}
```

Comment:
I think this would just show the warning for `accessLevel` so maybe should remove that prop to make sure it's the `imgKey` that prompts the warning